jerry-ext/include-private/section-utils.h

```diff
@@ -0,0 +1,69 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SECTION_UTILS_H
+#define SECTION_UTILS_H
+
+/**
+ * Declare references to a section that contains an array of items.
+ *
+ * @param name the name of the section (without quotes)
+ * @param type the type of the elements stored in the array
+ *
+ * This macro needs to be placed before any occurrence of ITERATE_OVER_SECTION, for a given section.
+ */
+#ifdef __MACH__
+#define DECLARE_SECTION(name, type)                                  \
+extern type __start_ ## name[] __asm("section$start$__DATA$" #name); \
+extern type __stop_ ## name[] __asm("section$end$__DATA$" #name);
```

Comment:
Ah, I see, thanks.

> So, does that mean that, for the Jerry heap it doesn't use a section on OSX?
 I guess so. I don't think it matters much when running on OSX.
I think it's probably intended for use on tiny systems where it may matter where you put the heap (you may have different types of RAM, switch of certain regions of RAM, etc.)